ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1546,60 +1548,66 @@ void handleResponse(ResponseInfo responseInfo, PutResponse putResponse) {
         logger.debug("{}: No matching request found for {}", loggingContext, correlationId);
         return;
       }
-      long requestLatencyMs = time.milliseconds() - chunkPutRequestInfo.startTimeMs;
-      routerMetrics.routerRequestLatencyMs.update(requestLatencyMs);
-      routerMetrics.getDataNodeBasedMetrics(chunkPutRequestInfo.replicaId.getDataNodeId()).putRequestLatencyMs.update(
-          requestLatencyMs);
       boolean isSuccessful;
       TrackedRequestFinalState putRequestFinalState = null;
-      if (responseInfo.getError() != null) {
-        logger.debug("{}: PutRequest with response correlationId {} timed out for replica {} ", loggingContext,
-            correlationId, chunkPutRequestInfo.replicaId.getDataNodeId());
-        setChunkException(new RouterException(
-            "Operation timed out because of " + responseInfo.getError() + " at DataNode " + responseInfo.getDataNode(),
-            RouterErrorCode.OperationTimedOut));
+      if (responseInfo.isQuotaRejected()) {
+        logger.debug("{}: PutRequest with response correlationId {} was rejected because quota was exceeded.",
+            loggingContext, correlationId);
+        setChunkException(new RouterException("QuotaExceeded", RouterErrorCode.TooManyRequests));
         isSuccessful = false;
-        putRequestFinalState = TrackedRequestFinalState.FAILURE;
+        putRequestFinalState = TrackedRequestFinalState.QUOTA_REJECTED;
```

Comment:
we should probably has a processQuotaRejectedResponse method here.

Reply:
done.